src/main/java/bio/terra/service/dataset/flight/ingest/IngestJsonFileSetupStep.java

```diff
@@ -0,0 +1,82 @@
+package bio.terra.service.dataset.flight.ingest;
+
+import bio.terra.common.Column;
+import bio.terra.model.IngestRequestModel;
+import bio.terra.model.TableDataType;
+import bio.terra.service.configuration.ConfigurationService;
+import bio.terra.service.dataset.Dataset;
+import bio.terra.service.dataset.exception.IngestFailureException;
+import bio.terra.service.filedata.google.gcs.GcsPdao;
+import bio.terra.stairway.FlightContext;
+import bio.terra.stairway.Step;
+import bio.terra.stairway.StepResult;
+import bio.terra.stairway.StepStatus;
+import bio.terra.stairway.exception.RetryException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.stream.Collectors;
+
+public class IngestJsonFileSetupStep implements Step {
+
+  private final GcsPdao gcsPdao;
+  private final Dataset dataset;
+  private final ObjectMapper objectMapper;
+  private final ConfigurationService configurationService;
+
+  public IngestJsonFileSetupStep(
+      GcsPdao gcsPdao,
+      ObjectMapper objectMapper,
+      Dataset dataset,
```

Comment:
I assume this works, but is it common to store the dataset in a step directly? Most other steps I've seen generate the dataset in the step use the ID and `DatasetService.retrieve()`, such as `DeleteDatasetDeleteStorageAccountsStep`, `DeleteDatasetPrimaryDataStep`, etc. Search for calls to `DatasetService.retrieve()` to see them all.

Not sure why we would avoid storing `Dataset` directly in a step but it looks like it's been our usual practice. Maybe it's large in its serialized form? Or, if its representation changes over time that could affect deserialization of `Dataset` but wouldn't affect deserializing the ID and looking it up.

Reply:
I'm really not sure. It looks like we use both conventions and it _seems_ more efficient to just pass the dataset object around instead of making yet another round trip to the database for each step just to get it. For flights like `DatasetIngestFlight`, the dataset isn't being modified at all, and the dataset is locked, so I don't think there's any danger in passing it around.

Reply:
Maybe this is a topic to discuss at standup. There is definitely a mix of forms here. One case I just looked at, `IngestFileMakeBucketLinkStep`, stores `Dataset` but internally only uses the id...

Reply:
We're gonna make a tech-debt ticket to harmonize what we do in all flights

Reply:
https://broadworkbench.atlassian.net/browse/DR-2069